### DISCUSSION TITLE
FileGlobLivestream Using 100% of all CPU Cores

### DISCUSSION THREAD

**Discussion by kylejbrk** [scivision/PyLivestream#95] [Q&A]
**Describe the bug**
When I start streaming it works fine, but after about a minute (maybe when going to the next file?) my CPU usage maxes out and causes the stream to buffer.

The only setting I changed in the pylivestream.ini is the preset to ultrafast, which didn't seem to help.

I'm also getting this error when starting the stream. Not sure if its related:
```

ERROR:root:FPS not available: /home/kyle/Documents/myStream/audio/Aviscerall - Nostalgia Infinite - 04 Scattered Dreams.mp3. Is it a video/audio file?

 ffmpeg -loglevel error -re -filter_complex movie=/home/kyle/Documents/myStream/video/publicDomainEarth.mp4:loop=0,setpts=N/FRAME_RATE/TB -i /home/kyle/Documents/myStream/audio/Aviscerall - Nostalgia Infinite - 04 Scattered Dreams.mp3 -codec:v libx264 -pix_fmt yuv420p -preset ultrafast -b:v 800k -r 30.0 -g 60.0 -codec:a aac -b:a 128k -ar 44100 -maxrate 800k -bufsize 400k -strict experimental -f flv rtmp://a.rtmp.youtube.com/live2/bju6-a2q1-cf73-61z8-619f 
```

**System Parameters**
`FileGlobLivestream ~/Documents/myStream/audio/ youtube -glob "*.mp3" -image ~/Documents/myStream/video/publicDomainEarth.mp4 -shuffle -loop`

* Operating system: Ubuntu
* FFmpeg version: 4.3.1

**Comment by scivision** [comment]
This is probably an issue with the media file or FFmpeg. I would try copy-pasting the command line you've noted above without using PyLivestream to see if the same issue happens.

**Comment by scivision** [comment]
I hear from a few users over time (out of ~ million users) that they have connection or CPU usage issues. I'm not sure as FFmpeg is doing all the work. I would try copy and pasting the command line and looking at FFmpeg docs/forums and try tweaking settings or try a different computer or network.
